src/Arcus.Observability.Tests.Unit/Correlation/DefaultCorrelationInfoAccessorTests.cs

```diff
@@ -14,24 +14,21 @@ public async Task SetCorrelationInfo_Twice_UsesMostRecentValue()
             var firstOperationId = $"operation-{Guid.NewGuid()}";
             var secondOperationId = $"operation-{Guid.NewGuid()}";
             var transactionId = $"transaction-{Guid.NewGuid()}";
-            await SetCorrelationInfo(firstOperationId, transactionId);
+            await SetCorrelationInfoAsync(firstOperationId, transactionId);
 
             // Act
-            await SetCorrelationInfo(secondOperationId, transactionId);
+            await SetCorrelationInfoAsync(secondOperationId, transactionId);
 
             // Assert
             CorrelationInfo correlationInfo = DefaultCorrelationInfoAccessor.Instance.GetCorrelationInfo();
             Assert.Equal(secondOperationId, correlationInfo.OperationId);
             Assert.Equal(transactionId, correlationInfo.TransactionId);
         }
 
-#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
-        // Disabled the warning since declaring this method as async was on-purpose
-        private async Task SetCorrelationInfo(string operationId, string transactionId)
-#pragma warning restore CS1998 // Async method lacks 'await' operators and will run synchronously
+        private async Task SetCorrelationInfoAsync(string operationId, string transactionId)
         {
-            DefaultCorrelationInfoAccessor.Instance.SetCorrelationInfo(
-                new CorrelationInfo(operationId, transactionId));
+            await Task.Run(() => DefaultCorrelationInfoAccessor.Instance.SetCorrelationInfo(
+                new CorrelationInfo(operationId, transactionId)));
         }
```

Comment:
This test case was made because we had a problem with setting our correlation in an asynchronously setting. It really is a unit test to secure this fix.
Also note that this is testing deprecated functionality.

Reply:
Ah.  It would have been good if the name of the test described that we're specifically testing this in an async context.

Reply:
Added `...InAsyncContext` to the test's name.